packages/sqlite/src/lib/SQLiteProvider.ts

```diff
@@ -569,60 +569,70 @@ export class SQLiteProvider<StoredValue = unknown> extends JoshProvider<StoredVa
   }
 
   public [Method.Random](payload: Payload.Random<StoredValue>): Payload.Random<StoredValue> {
-    const { count, duplicates } = payload;
+    const { count, unique } = payload;
     const size = this.handler.size();
 
-    if (size === 0) return { ...payload, data: [] };
-    if (size < count) {
+    if (unique && size < count) {
       payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidCount, method: Method.Random }, { size }));
 
       return payload;
     }
 
+    if (size === 0) {
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Random }, { unique, count }));
+
+      return payload;
+    }
+
     payload.data = [];
 
     const keys = this.handler.keys();
 
-    if (duplicates) {
-      while (payload.data.length < count) {
-        const key = keys[Math.floor(Math.random() * size)];
-
-        payload.data.push(this.handler.get(key)!);
-      }
-    } else {
+    if (unique) {
       const randomKeys = new Set<string>();
 
       while (randomKeys.size < count) randomKeys.add(keys[Math.floor(Math.random() * keys.length)]);
```

Comment:
Same optimization question

Reply:
Seems the same as the other 2 sql languages